gluten-substrait/src/main/scala/org/apache/gluten/extension/columnar/PushDownInputFileExpression.scala

```diff
@@ -125,9 +125,9 @@ object PushDownInputFileExpression {
           partitionPruningPred = child.partitionPruningPred,
           prunedOutput = child.prunedOutput
         )(child.session)
-      case p @ ProjectExec(projectList, child: BatchScanExecTransformer)
+      case p @ ProjectExec(projectList, child: BatchScanExecTransformerBase)
           if projectList.exists(containsInputFileRelatedExpr) =>
-        child.copy(output = p.output.asInstanceOf[Seq[AttributeReference]])
+        child.withNewOutput(p.output.asInstanceOf[Seq[AttributeReference]])
```

Comment:
Would you like to help check whether we can share this approach with Delta?

https://github.com/apache/incubator-gluten/blob/main/gluten-delta/src/main/scala/org/apache/gluten/extension/DeltaPostTransformRules.scala

Reply:
Would you open an issue for this? Thanks!

We may want to end up with a common solution for different lake formats.

Reply:
https://github.com/apache/incubator-gluten/issues/10840